lib/OpenQA/ServerStartup.pm

```diff
@@ -155,4 +156,35 @@ sub setup_logging {
     $OpenQA::Utils::app = $app;
 }
 
+# Update config definition from plugin requests
+sub update_config {
+    my ($config, @namespaces) = @_;
+    return unless exists $config->{ini_config};
+
+    # Filter out what plugins are loaded from the used namespaces
+    foreach my $plugin (loaded_plugins(@namespaces)) {
+
+        # We take config only if the plugin has the method declared
+        next unless ($plugin->can("configuration_fields"));
+
+        # If it is a Mojo::Base class, it requires to be instantiated
+        # because the attributes are not populated until creation.
+        my $fields
+          = UNIVERSAL::isa($plugin, "Mojo::Base") ?
+          do { $plugin->new->configuration_fields() }
```

Comment:
can you enlighten me on this syntax? looks quite unusual to have a `do {}` inside a ternary operator. Wouldn't an explicit `if/else` be easier to read here?